meson.build

```
@@ -134,6 +134,16 @@ if get_option('static_analyzer').enabled()
   add_project_arguments(['-fanalyzer'], language: 'c')
 endif
 
+if get_option('sections').enabled()
+  add_project_arguments(['-ffunction-sections'], language: 'c')
+  link_args = [
+    '-Wl,--gc-sections',
+    '-Wl,--print-gc-sections',
+  ]
+else
+  link_args = []
+endif
```

nit: I suggest the style below as we may conditionally add other linker flags in the future.

```suggestion
link_args = []
if get_option('sections').enabled()
  add_project_arguments(['-ffunction-sections'], language: 'c')
  link_args += [
    '-Wl,--gc-sections',
    '-Wl,--print-gc-sections',
  ]
endif
```